lib/modules/manager/npm/extract/pnpm.ts

```diff
@@ -10,9 +11,15 @@ import {
   localPathExists,
   readLocalFile,
 } from '../../../../util/fs';
+import { regEx } from '../../../../util/regex';
 import type { PackageFile } from '../../types';
+import type { PnpmLockFile } from '../post-update/types';
 import type { NpmManagerData } from '../types';
-import type { PnpmWorkspaceFile } from './types';
+import type { LockFile, PnpmWorkspaceFile } from './types';
+
+function isPnpmLockfile(obj: any): obj is PnpmLockFile {
+  return is.plainObject(obj);
```

Comment:
```suggestion
  return is.plainObject(obj) && 'lockfileVersion' in obj;
```
some more validation, we can later use zod to ensure

Reply:
I was also looking into zod, do remember me when you create an issue😏